unyt/unit_object.py

```diff
@@ -711,22 +718,26 @@ def latex_representation(self):
 #
 
 
+# map from dimensions in one unit system to dimensions in other system,
+# canonical unit to convert to in that system, and floating point
+# conversion factor
 em_conversions = {
-    "C": ("esu", 0.1*speed_of_light_cm_per_s, "cgs"),
-    "T": ("gauss", 1.0e4, "cgs"),
-    "Wb": ("Mx", 1.0e8, "cgs"),
-    "A": ("statA", 0.1*speed_of_light_cm_per_s, "cgs"),
-    "V": ("statV", 1.0e-8*speed_of_light_cm_per_s, "cgs"),
-    "ohm": ("statohm", 1.0e9/(speed_of_light_cm_per_s**2), "cgs"),
-    "esu": ("C", 10.0/speed_of_light_cm_per_s, "mks"),
-    "Fr": ("C", 10.0/speed_of_light_cm_per_s, "mks"),
-    "statC": ("C", 10.0/speed_of_light_cm_per_s, "mks"),
-    "gauss": ("T", 1.0e-4, "mks"),
-    "G": ("T", 1.0e-4, "mks"),
-    "Mx": ("Wb", 1.0e-8, "mks"),
-    "statA": ("A", 10.0/speed_of_light_cm_per_s, "mks"),
-    "statV": ("V", 1.0e8/speed_of_light_cm_per_s, "mks"),
-    "statohm": ("ohm", speed_of_light_cm_per_s**2*1.0e-9, "mks"),
+    dims.charge_mks: (dims.charge_cgs, "esu", 0.1*speed_of_light_cm_per_s),
+    dims.charge_cgs: (dims.charge_mks, "C", 10.0/speed_of_light_cm_per_s),
+    dims.magnetic_field_mks: (dims.magnetic_field_cgs, "gauss", 1.0e4),
+    dims.magnetic_field_cgs: (dims.magnetic_field_mks, "T", 1.0e-4),
+    dims.current_mks: (
+        dims.current_cgs, "statA", 0.1*speed_of_light_cm_per_s),
+    dims.current_cgs: (dims.current_mks, "A", 10.0/speed_of_light_cm_per_s),
+    dims.electric_potential_mks: (
+        dims.electric_potential_cgs, "statV",
+        1.0e-8*speed_of_light_cm_per_s),
+    dims.electric_potential_cgs: (
+        dims.electric_potential_mks, "V", 1.0e8/speed_of_light_cm_per_s),
+    dims.resistance_mks: (
+        dims.resistance_cgs, "statohm", 1.0e9/(speed_of_light_cm_per_s**2)),
+    dims.resistance_cgs: (
+        dims.resistance_mks, "ohm", 1.0e-9*speed_of_light_cm_per_s**2)
 }
```

Comment:
I actually tried this before and it didn't work--and it was because two of these dimensions actually reduced to the same symbol, so one of them got overwritten in the dictionary. But now I can't remember which two....

Reply:
I think it was Maxwell. I decided that magnetic field strength was more important so I removed support for automatic conversions of magnetic flux.

Reply:
Ah, fair enough.